### ISSUE TITLE
Settings for "Stick post" and "Unstick post"

### ISSUE THREAD

**Issue by stevejburge** [publishpress/PublishPress-Future#866]
Sorry, these aren't very clear:
![Screenshot 2024-07-12 at 8 58 50 AM](https://github.com/user-attachments/assets/428bf0e9-74ee-4352-ae04-de68800ed3bb)

**Comment by andergmartins** [comment]
This is probably related to #716 because the same "Stick post" should be able to execute the task for the post it receives from the fired trigger. The problem it is currently expecting a specific variable, from a specific trigger. That should be optional. 

We should be able to set an expression in the Post field on Stick post (and other similar actions) saying it should use `onPostUpdated1.postAfter OR onSavePost1.post OR onSavePost2.post`.

I'm just not sure yet what is the best UI or UX or solving this.

**Comment by andergmartins** [comment]
@stevejburge to fix this specific issue, I suggest we display the step slug in the diagram nodes and start using the variable name instead of a label on setting fields. This will keep that more consistent with the list of variables we display when Advanced Settings is enabled, and works as a preparation for #716 when we will implement support to expressions.



![Image](https://github.com/user-attachments/assets/82abae7d-1124-4cb4-adeb-aa8d2497bca4)

**Comment by stevejburge** [comment]
Thanks @andergmartins. I guess we don't need any dropdown options show here? Sticking or Unsticking a post is a simple yes / no choice.

**Comment by andergmartins** [comment]
@stevejburge for simple workflows probably not (but requires some changes in how we manage data inside the workflow execution), but for more complex worflows (thinking about the future) it would be good to have an option to customize that.

I will try to simplify this, showing less options by default, but still giving the option to customize the node input.

**Comment by andergmartins** [comment]
@stevejburge I think we can improve this, but it will require some work. 
I suggest we move this to 3.4.4 to not block a release of 3.4.3 next week with other bug fixes.

**Comment by stevejburge** [comment]
Sounds good to me

**Comment by andergmartins** [comment]
> Thanks [@andergmartins](https://github.com/andergmartins). I guess we don't need any dropdown options show here? Sticking or Unsticking a post is a simple yes / no choice.

Maybe related to #912?

**Comment by stevejburge** [comment]
Yes, please. Let's not force the users to choose. Let's go with the obvious choice as the default.

**Comment by andergmartins** [comment]
@stevejburge I'm not sure yet what is the best path to take for fixing this one, and I see the source of confusion there.

Any setting that will make reference to a variable passed by previous steps will require a very specific reference to the variable. That means that variables are deeply connected to their source.

It a trigger "step1" outputs a post variable, it is named "step1.post".
For a trigger "step2", the outputted variable is "step2.post". 

But you don't see that there because we are display the variable's label instead of it's name.

Child steps, when making reference to a post, will have to select a specific variable, not just "post". So on its settings we will have to fill with "step1.post" or "step2.post". We can't set if for both.

## Fix 1

One possible fix is not allowing to have multiple triggers for the same branch or step.

**Cons**

* Backward compatibility with workflows that possibly already use multiple triggers.

## Fix 2

The other possible fix is keep the multiple triggers for the same step, but provide a way to select multiple variables, and it will use the current available one. This forces us to remove ambiguity showing the variable name instead of the label.

**Cons**

* We still need to explain to the user the concept I explained on this comment.

---

I will take another look at this one with fresh eyes tomorrow.

**Comment by stevejburge** [comment]
Thanks @andergmartins. I admit to not understanding this issue fully. Are we over-complicating this? 

When it comes to "Stick Post", I'd assume this referred only to "Post After Update".

**Comment by andergmartins** [comment]
@stevejburge maybe for most situations it seems like an over-complication.
But that is very important for the Conditional step.
Some decisions in the workflow can be taken depending on how the post was updated. 
And how to know what changed, if we don't provide the before and after state of the post?

**Comment by stevejburge** [comment]
@andergmartins OK, in that case, let's just make sure to set a logical default. Which option will make sense for 90% of users? Let's choose that automatically.

**Comment by stevejburge** [comment]
OK, thanks. In that case, perhaps let's postpone this old issue. We'll see what customer feedback comes in.
